proposals/cloud_detection.json

```diff
@@ -0,0 +1,65 @@
+{
+    "id": "cloud_detection",
+    "summary": "Cloud detection",
+    "description": "Detects clouds and cloud shadow in optical imagery. By default, the cloud mask is applied to the input data cube. Set `apply_mask` to `false` to add the cloud mask as a new band.",
+    "categories": [
+        "cubes",
+        "optical"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "description": "Data cube containing multi-spectral optical top of atmosphere reflectances on which to perform cloud detection.",
+            "name": "data",
+            "schema": {
+                "subtype": "raster-cube",
+                "type": "object"
+            }
+        },
+        {
+            "description": "The cloud detection method to use.\n\nThe supported algorithms can vary between back-ends. Set to `null` (the default) to allow the back-end to choose, which will improve portability, but reduce reproducibility.",
+            "name": "method",
+            "optional": true,
+            "default": null,
+            "schema": {
+                "type": [
+                    "string",
+                    "null"
+                ],
+                "enum": [
+                    "sen2cor",
+                    "fmask",
+                    "s2cloudless"
+                ]
+            }
+        },
+        {
+            "description": "By default, the cloud mask is applied to the input data cube. If set to `false`, the cloud mask information is added as one or more additional bands. Some implementations may add additional bands.",
```

Comment:
I'm wondering whether this should be two processes as the behavior is somewhat different and in one case the data cube dimension labels change while in the other the data cube stays exactly as it is.

Reply:
So: cloud_detection and cloud_mask as two processes.
cloud_detection could return only cloud mask bands?

cloud_mask can be a user defined process, defined in terms of cloud_detection?

What about method specific parameters?
- common denominator
- custom backend-specific object (e.g. context parameter)

Reply:
Just discussed with Peter: Why do we actually need cloud_mask? It feels like we just can use our regular mask function. Any objections to remove/not have cloud_mask and clouc_detection just return a new data cube with the masks?

Reply:
Made a new attempt at making cloud_detection more general, more useful and easier to use. It doesn't allow applying the masks anymore, instead we propose to use the mask process. Please see the new version of the process and give feedback. We are pretty sure it's not the last iteration, e.g. we need to discuss which aspects to name (water vapor?) and how to name the dimension labels.